src/java.base/share/classes/jdk/crac/impl/BlockingOrderedContext.java

```diff
@@ -0,0 +1,60 @@
+package jdk.crac.impl;
+
+import jdk.crac.CheckpointException;
+import jdk.crac.Context;
+import jdk.crac.Resource;
+import jdk.crac.RestoreException;
+import jdk.internal.crac.LoggerContainer;
+
+public class BlockingOrderedContext<R extends Resource> extends OrderedContext<R> {
+    private boolean checkpointing = false;
+
+    // This method has particularly verbose name to stick out in thread dumps
+    // when the registration leads to a deadlock.
+    private void waitWhileCheckpointIsInProgress(R resource) {
+        if (Thread.currentThread().isInterrupted()) {
+            LoggerContainer.debug(Thread.currentThread().getName() + " not waiting in " + this +
+                " to register " + resource + "; the thread has already been interrupted.");
+            // We won't cause IllegalStateException because this is not an unexpected state
+            // from the point of CRaC - it probably tried to register some code before.
+            throw new RuntimeException("Interrupted thread tried to block in registration of " + resource + " in " + this);
```

Comment:
The use of `this` in the exception relies on naming the context and the `toString()` method for easy identification. Since you've removed these it will show only class type rather than Global Context/custom name/JDK resource priority.

Reply:
The name complicates the interface and from the stack trace the context will be completely evident.